qoqo/tests/integration/operations/three_qubit_gate_operations.rs

```diff
@@ -0,0 +1,473 @@
+// Copyright © 2021-2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ndarray::Array2;
+use num_complex::Complex64;
+use numpy::PyArray2;
+use qoqo_calculator_pyo3::CalculatorFloatWrapper;
+
+use std::collections::HashMap;
+
+use pyo3::prelude::*;
+
+use qoqo::{
+    operations::{
+        convert_operation_to_pyobject, ControlledControlledPauliZWrapper,
+        ControlledControlledPhaseShiftWrapper,
+    },
+    CircuitWrapper,
+};
+use qoqo_calculator::CalculatorFloat;
+use roqoqo::{operations::*, Circuit, RoqoqoError};
+
+use test_case::test_case;
+
+// helper function to convert CalculatorFloat into a python object
+fn convert_cf_to_pyobject(
+    py: Python,
+    parameter: CalculatorFloat,
+) -> &PyCell<CalculatorFloatWrapper> {
+    let parameter_type = py.get_type::<CalculatorFloatWrapper>();
+    match parameter {
+        CalculatorFloat::Float(x) => parameter_type
+            .call1((x,))
+            .unwrap()
+            .downcast::<PyCell<CalculatorFloatWrapper>>()
+            .unwrap(),
+        CalculatorFloat::Str(x) => parameter_type
+            .call1((x,))
+            .unwrap()
+            .downcast::<PyCell<CalculatorFloatWrapper>>()
+            .unwrap(),
+    }
+}
```

Comment:
Should this function be moved to the mod.rs and imported in all the files that use it?